### PR TITLE
[DK] repo cleanup: type errors, remove lib-based histogram marker, etc

### PR DESCRIPTION
**Pull request by moontrip**
This is for cleaning repo: fixed various type errors; removed library-based histogram marker (plotly.js & high-chart); remove knobs and set default values for a couple of variable (e.g., border color, width, etc.); removed unnecessary tsx files (BarChart, BarChartLegend, and MapVeuLegendSampleDropdown); removed those library packages and their type definitions, etc.

### CODE REVIEW

**Review by bobular — COMMENTED** [review]
Looks good DK.

I would recommend some more renaming. Simplify and shorten.  e.g. SVGDonutMarker.tsx -> DonutMarker.tsx and RealHistogramMarkerSVGnoShadow.tsx -> ChartMarker.tsx

Also be careful with capitalisation (MapVeu->MapVEu)

Ignore anything sidebar-related at the moment (this will be tidied up soon - see MapVEu blog and new ticket https://redmine.apidb.org/issues/42968)

Should we remove (or comment out, lol) the <Tooltip> elements now that we are done testing?  They were really only for debugging.

Thanks!

File: src/map/RealHistogramMarkerSVGnoShadow.tsx [hunk 1]
```diff
@@ -100,74 +90,36 @@ export default function RealHistogramMarkerSVGnoShadowAnim(props: HistogramMarke
...
-        ></BarChart>,
-        document.getElementById(id)
-      );
-
-      // Deconstruct the chart on marker derender
-      return () => {
-        const el = document.getElementById(id);
-      if (el) ReactDOM.unmountComponentAtNode(el);
-      }
-    });
```
[the remark this answers is not in the document]
Comment (moontrip) [reply src/map/RealHistogramMarkerSVGnoShadow.tsx:162]: Thank you for your review and suggestion, @bobular ! I will do that and commit again after testing

**Comment by moontrip** [comment]
> Looks good DK.
> 
> I would recommend some more renaming. Simplify and shorten. e.g. SVGDonutMarker.tsx -> DonutMarker.tsx and RealHistogramMarkerSVGnoShadow.tsx -> ChartMarker.tsx
> 
> Also be careful with capitalisation (MapVeu->MapVEu)
> 
> Ignore anything sidebar-related at the moment (this will be tidied up soon - see MapVEu blog and new ticket https://redmine.apidb.org/issues/42968)
> 
> Should we remove (or comment out, lol) the elements now that we are done testing? They were really only for debugging.
> 
> Thanks!

Thank you for your review and suggestion, @bobular ! I will do that and commit again after testing

**Comment by moontrip** [comment]
@bobular I have made changes following your comments~

**Pull request merged**